lmdeploy/serve/openai/api_client.py

```diff
@@ -7,11 +7,15 @@
 from lmdeploy.utils import get_logger
 
 
-def get_model_list(api_url: str):
+def get_model_list(api_url: str, headers: dict = None):
     """Get model list from api server."""
-    response = requests.get(api_url)
+    response = requests.get(api_url, headers=headers)
+    if not response.ok:
+        logger = get_logger('lmdeploy')
+        logger.error(f'Failed to get the model list: {api_url} returns {response.status_code}')
+        return None
     if hasattr(response, 'text'):
-        model_list = json.loads(response.text)
+        model_list = response.json()
         model_list = model_list.pop('data', [])
         return [item['id'] for item in model_list]
     return None
```

Comment:
there are two `return None`. May make it cleaner.

Reply:
Fixed